src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -214,21 +223,27 @@ protected Object getObject(BsonValue o, int columnType) throws SQLException {
     @Override
     public Object getObject(int columnIndex) throws SQLException {
         BsonValue out = getBsonValue(columnIndex);
+        // If the value is Null, no need to try to convert to a Java Object
```

Comment:
This check is unnecessary since we are calling `getObject(value, columnType)` which also checks it.

Reply:
Yes, it's only here to saves us a call rsMetaData.getColumnType which is useless if the data is null.

Reply:
I went back and checked the code for getColumnType. It's calling a few getters on already set variables, so should not be too expensive. Seems reasonable to leave checkNull for `getObject(value, columnType)` as you pointed out :)